api/utils/addr.go

```diff
@@ -0,0 +1,28 @@
+package utils
+
+import (
+	"net"
+	"strings"
+)
+
+// IsLoopback returns 'true' if a given hostname resolves to local
+// host's loopback interface
+func IsLoopback(host string) bool {
+	if strings.Contains(host, ":") {
+		var err error
+		host, _, err = net.SplitHostPort(host)
+		if err != nil {
+			return false
+		}
+	}
+	ips, err := net.LookupIP(host)
+	if err != nil {
+		return false
+	}
+	for _, ip := range ips {
+		if ip.IsLoopback() {
+			return true
+		}
+	}
```

Comment:
Could there be any situations where packets might not be sent over the loopback interface if multiple IPs are returned? 

Go's docs (https://pkg.go.dev/net#Dial) say:
> When using TCP, and the host resolves to multiple IP addresses, Dial will try each IP address in order until one succeeds.

I could maybe see this being problematic if an attacker got control of the host's DNS.

Reply:
I'm not sure, it might be reasonable for a host to resolve to both a loopback address and a public facing address. It doesn't seem to be common at least, my Linux box only resolves its public-facing IPs (IPv4 and IPv6) by default, though I can define several in `/etc/hosts`.

Reply:
I'm going to err on the side of paranoia and say that a hostname refers to a loopback address if _everything_ it resolves to is the loopback.